src/support/rand.c

```diff
@@ -117,8 +118,17 @@ __wt_random_init_seed(WT_SESSION_IMPL *session, WT_RAND_STATE volatile *rnd_stat
      * Take the seconds and nanoseconds from the clock together with the thread ID to generate a
      * 64-bit seed, then smear that value using algorithm "xor" from Marsaglia, "Xorshift RNGs".
      */
-    M_W(rnd) = (uint32_t)ts.tv_sec ^ DEFAULT_SEED_W;
-    M_Z(rnd) = (uint32_t)ts.tv_nsec ^ DEFAULT_SEED_Z;
+    M_W(rnd) =
+      (uint32_t)ts.tv_sec ^ (uint32_t)WT_LEFT_CIRCULAR_SHIFT32(ts.tv_nsec, 29) ^ DEFAULT_SEED_W;
+    M_Z(rnd) =
+      (uint32_t)ts.tv_nsec ^ (uint32_t)WT_LEFT_CIRCULAR_SHIFT32(ts.tv_sec, 27) ^ DEFAULT_SEED_Z;
+/*
+ * Window's clock does not have a high enough resolution between each tick cycle. Perform an extra
+ * xor against the machine's timestamp counter.
+ */
+#ifdef _WIN32
+    rnd.v ^= __wt_rdtsc();
+#endif
```

Comment:
Consider xor-ing `__wt_rdtsc()` unconditionally, and change the comment: `Some system clocks do not have...`. It's a little more future-proof.

Reply:
I would rather not xor __wt_rdtsc() unconditionally yet. Linux systems seem to be doing okay with the current circular shift change, my opinion is that we only start thinking about it if mongodb has gotten failures regarding other machines as well.